apps/client/src/app/pages/portfolio/fire/fire-page.component.ts

```diff
@@ -149,6 +150,29 @@ export class FirePageComponent implements OnDestroy, OnInit {
       });
   }
 
+  public onRulesUpdated(event: UpdateUserSettingDto) {
+    this.isLoading = true;
+    this.dataService
+      .putUserSetting(event)
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe(() => {
+        this.dataService
+          .fetchPortfolioReport()
+          .pipe(takeUntil(this.unsubscribeSubject))
+          .subscribe((portfolioReport) => {
+            this.accountClusterRiskRules =
+              portfolioReport.rules['accountClusterRisk'] || null;
```

Comment:
This warning comes out when activating or deactivating a rule https://angular.dev/errors/NG0956. This is because when we update the rule, we fetch and update the DOM for all rules. However the `firePageComponent` doesn't have any clues of which rules belongs to which category, so reloading a single category or rule is impossible. I am trying to find a way to do so.

Reply:
What happens if you change `track rule` to `track rule.evaluation` [here](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/components/rules/rules.component.html#L16)?

If that works, it should finally be `track rule.key`, but this seems to be missing on `PortfolioReportRule`.

Reply:
Indeed, I wasn't aware of this `track` directive. I will work on adding the `key` so we reload only the correct key.

Reply:
Actually, `key` is present in `PortfolioReportRule`, so we can reload only the modified rule.